src/zarr/core/metadata/v2.py

```diff
@@ -327,7 +327,15 @@ def parse_fill_value(fill_value: Any, dtype: np.dtype[Any]) -> Any:
     """
 
     if fill_value is None or dtype.hasobject:
+        # Pass through None or if dtype is object
         pass
+    elif dtype.kind in "M":
```

Comment:
```suggestion
    elif dtype.kind in ("M", "m"):
```
handle timedelta64 as well